### DISCUSSION TITLE
Splash Screen while server is generating

### DISCUSSION THREAD

**Discussion by yuriburk** [vercel/next.js#15974] [Help] [answered]
How do I show a Splash Screen (any component actually) while server is generating the HTML to pages/Index (getStaticProps)?

I know that can be done with Router when navigate to another page, but It doesn't work on the home page (I want to show it on site loading). Any help it's appreciated. Thanks.

**Comment by nvdvrts** [comment]
+1

**Reply by stigkj** [reply]
@yuriburk @nvdvrts Not sure if you have found out, but this has been possible since v9.3 by using the fallback support in `getStaticPaths`/`router`.

Tl;dr: return `{..., fallback: true }` from `getStaticPaths` and any path not pre-rendered on build will have a `.isFallback` property on the router object (from `useRouter`) that is true. I.e. you could render a splash screen when this property is true.

Check https://nextjs.org/docs/basic-features/data-fetching#fallback-pages for more info.

**Reply by nvdvrts** [reply]
Very helpful! Is this possible with routing? `router.push('secondPage')`

**Reply by stigkj** [reply]
@nvdvrts Not sure if I understand your question, but I think that should work. I.e. you are on a page and want to load `secondPage` which should show a splash screen. Then doing `router.push('secondPage')` should load the page generated on build (the splash screen) while loading data from the client in the background and updating the page on the client when data is received.

**Comment by yuriburk** [comment, marked as answer]
Closing this thanks to @stigkj https://nextjs.org/docs/basic-features/data-fetching#fallback-pages
